lib/stream/multi.rb

```diff
@@ -46,7 +46,7 @@ def publisher_prefill
 
     if inviter = self.user.invited_by.try(:person)
       prefill << I18n.t("shared.publisher.new_user_prefill.invited_by")
-      prefill << "@{#{inviter.name} ; #{inviter.diaspora_handle}}!"
+      prefill << "@{#{inviter.diaspora_handle}}!"
```

Comment:
I thought this line would be enough to fix the bug. Is the usage of gon necessary or is it a refactor for a cleaner code?

Reply:
No, this change is needed to make gon work (`addPersonToMentions` only works with this syntax).

The main fix was in the [publisher.rb](
https://github.com/diaspora/diaspora/pull/7702/files#diff-a51abdb338000222bc972a44538e8b6bL15), because the `.plain_text` changed this prefill message to plain text and replaced the mention syntax with the plain text version (which then obviously doesn't work anymore, because it's only plain text and no mention).

But that only would fix it in the final post. The gon part is to make the preview work.